back_end/src/models/friend.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from typing import List
+
+from ..setup import db
+
+
+class Friend(db.Model):
+    """
+    @author: YixuanZ
+    """
+    __tablename__ = "Friends"
+
+    id = db.Column(db.Integer, primary_key=True)
+    user1_id = db.Column(db.Integer, db.ForeignKey('Users.id'), nullable=False)
+    user2_id = db.Column(db.Integer, db.ForeignKey('Users.id'), nullable=False)
+
+    def __init__(self, **kwargs):
+        super(Friend, self).__init__(**kwargs)
+
+    def to_json(self):
+        ret = {}
+        ret['id'] = self.id
+        ret['user1_id'] = self.user1_id
+        ret['user2_id'] = self.user2_id
+        return ret
+
+    def save(self):
+        db.session.commit()
+
+    @staticmethod
+    def add_friend(user1_id: int, user2_id: int) -> Friend:
+        f = Friend.query.filter_by(user1_id=user1_id,
+                                   user2_id=user2_id).first()
+        if f:
+            return None
+        f = Friend(user1_id=user1_id, user2_id=user2_id)
+        db.session.add(f)
+        f.save()
+        return f
+
+    @staticmethod
+    def is_friend(user1_id: int, user2_id: int) -> bool:
+        if Friend.get_freind_by_sender_and_receiver(user1_id=user1_id,
+                                                    user2_id=user2_id):
+            if Friend.get_freind_by_sender_and_receiver(user1_id=user2_id,
+                                                        user2_id=user1_id):
+                return True
+        return False
+
+    @staticmethod
+    def get_freind_by_sender_and_receiver(user1_id: int,
```

Comment:
wrong method name, but it has the same name as when you call it in apis/friend.py